reactive/containerd.py

```diff
@@ -124,14 +126,46 @@ def charm_status():
         status.blocked('Container runtime not available')
 
 
-def merge_custom_registries(custom_registries):
+def update_custom_tls_config(config_directory, registries):
+    """
+    Read registries config and write tls files to disk.
+
+    :param str config_directory: containerd config directory
+    :param List registries: juju config for custom registries
+    :return: None
+    """
+    for registry in registries:
+        for opt in ['ca', 'key', 'cert']:
+            file_b64 = registry.get('%s_file' % opt)
+            if file_b64:
+                try:
+                    file_contents = base64.b64decode(file_b64)
+                except (binascii.Error, TypeError):
+                    log(traceback.format_exc())
+                    log("{}:{} didn't look like base64 data... skipping"
+                        .format(registry['url'], opt))
+                    status.blocked(
+                        'Invalid base64 data in custom_registries: {}:{}_file'.format(
+                            registry['url'], opt))
```

Comment:
```suggestion
```

Reply:
@addyess sorry to flip-flop on you, but I now think we should go with your initial impl and handle config validation separately.  Blocking on invalid tls is still probably a good idea, but I think we'll be better served by blocking on any invalid custom_registry entry. That's out of the scope of this PR, so I've opened this to track the effort:

https://bugs.launchpad.net/charm-containerd/+bug/1881173